### DISCUSSION TITLE
C3ETH Operations; Translation Compensation Structure

### DISCUSSION THREAD

**Discussion by PardusEidolon** [C3ETH/c3eth.github.io#90] [Ideas]
# Translation Compensation Structure

Compensation should be tied to the budget for translation and the T-Shirt size estimate. 

I suggest we first work with four T-Shirt sizes **Xtra-Small (XS)**, **Small (S)**, **Medium (M)**, **Large (L)** and reduce the amount compensated for each. Taking @YanTirta word count ranges and prior reported effort as a starting point: 

1. **XS**: 0 ADA. These are useful getting started tasks (approximately 0 to 200 words).
2. **S**: 5 ADA (approximately 200 to 400 words).
3. **M**: 15 ADA (approximately 400 to 800 words).
4. **L**: 25 ADA (approximately 800 and above)

We should keep in mind that word count is only indicative. A very technical proposal short in word count can take much more effort than a high word-count proposal that is 100% free of jargon. Likewise, budgets will dictate what we can spend on translation and language.

Also, don't forget we can issue a badge for doing translations. So, for instance, the XS tasks may not pay anything individually, but if you complete five of them, you can get an ETH Translation badge.

I have also added a wiki page in the `wiki tab` for this repository It just needs its details to be filled out.

here in _fig 1.1_ have added a feature to choose which t-shirt size. Depending on big the task is a member can choose a t-shirt size on an open issue displayed in _fig 2.2_

![t-shirt-size](https://user-images.githubusercontent.com/38515818/149606898-50d038cc-a32b-4983-9ed5-4c330f37e8f9.png)
_fig 1.1_

![t_shirt_sizes](https://user-images.githubusercontent.com/38515818/149610122-ec893bfe-9ceb-4d74-ae03-f4c2467ce5b1.gif)
_fig 1.2_

**Comment by wolstaeb** [comment]
@PardusEidolon Using the new Github Project Boards you can set up a custom attribute for holding the T-Shirt estimate of a Translation.

**Comment by YanTirta** [comment]
Great job @PardusEidolon ! Have already the gist of this t-shirt system. OOT question, both _fig 1.1_ and _fig 1.2_ are just regular *.gifs right?

**Reply by PardusEidolon** [reply]
`fig 1.1` is a regular image of the project board and  `fig 1.2` is the gif :-)

**Comment by zicozibu** [comment]
@PardusEidolon I've followed the steps in the gif, until the translation page, but found nothing under Projects.
I wonder if that's because I'm not in the Assignees List?
![Screen Shot 2022-01-19 at 18 48 59](https://user-images.githubusercontent.com/11385285/150124390-38a43f35-d4c4-4a0d-ab7e-5c07292ce0e4.png)

**Reply by YanTirta** [reply]
That might be the issue but I'm not 100% certain. @PardusEidolon , would love your input on this.

**Reply by PardusEidolon** [reply]
It's because there isn't a project assigned to the issue, an oversight on my part. which issue are you referring to here?

**Reply by YanTirta** [reply]
Pardon my language reference. The use of issue here, is in terms of the issue that @zicozibu is facing, where he can not contribute in determining which size a translation item should be.

**Comment by zicozibu** [comment]
@PardusEidolon some questions/suggestions perhaps:

1. On the XS tasks, is it after 5 tasks the translator will get rewarded with the total number of words (and go with either S, M, L class)? Or will he/she got badge only? 

2. Isn't issuing badges only add complexity in the whole process?

3. Would it be better for all of us to go with simple yet accountable pipeline?

4. By 'simple', I would suggest that we go with this T-Shirt sizes (based on word count), and then put in another difficulty factor (let's say: normal, difficult, terrifying) to actually "upgrade" the T-Shirt sizes.

For example: 
normal : +0 (no upgrade)
difficult: +1 (upgrade to next size)
terrifying: +2 (upgrade to 2 next size)

- case 1 : 350 words, normal --> would be in S size, with no upgrade --> final size: S
- case 2: 300 words, difficult --> would be in S size, with +1 upgrade --> final size: M
- case 3: 150 words, terrifying --> would be in XS size, with +2 upgrade --> final size: M

or even more simple: 
just take 2 difficulty levels:
- normal : without extra reward
- hard: extra reward of S size (+5 ADA)
so back in the cases:
- case 1 : 350 words, normal --> would be in S size, with no upgrade --> final reward: S
- case 2: 300 words, hard --> would be in S size, with extra S reward --> final reward: S+S

**Reply by zicozibu** [reply]
I think if we can agree on that, we can start counting the reasonable reward for each of the T-Shirt sizes, by comparing: 

**the number of words to be translated (_target_)** VS **the number of ADA (_budget_)**

say, we have **Target** of total 50,000 words (total from say 100 articles), and our **Budget** is at 1,000 ADA, that means per word we would have 1,000 ADA/50,000 words = 0,02 ADA/word

The 0,02 ADA/word is for reference, so we can go for the T-Shirt sizes compensation by using the average of the sizes range. If we take on your example:
1. XS: 1 to 200 words ~~ 100 words x 0,02 ADA =  2 ADA  (the original plan was 0 ADA)
2. S: 201 to 400 words ~~ 300 words x 0,02 ADA = 6 ADA  (the original plan was 5 ADA)
3. M: 401 to 800 words ~~ 600 words x 0,02 ADA = 12 ADA  (the original plan was 15 ADA)
4. L:  801 and above, say 1600 words) ~~ 1200 words x 0,02 ADA = 24 ADA  (the original plan was 25 ADA)
_(please note that all the numbers above is for illustration purpose only)_

I think this is quite fair yet will be interesting enough for the translators to pick any task size, and simple to maintain (we can always change dynamically according the **Target** and **Budget**)

**Reply by zicozibu** [reply]
If we can agree on this, very practical steps can be done: 
1. count the **Target**
2. define the **Rate** (ADA per word) _(I assume the **Budget** is set/known)_, with everyone agree on the **Rate**
3. publish to the community
4. get the result and track record for the report for Catalyst
... and
5. get ready for proposal for the next Fund 😃

**Reply by YanTirta** [reply]
Great input @zicozibu . Using the word count as a foundation, then adding to it the T-Shirt model by using difficulty metrics to upgrade or downgrade a category I think is crystal clear and logical.

**Reply by PardusEidolon** [reply]
@YanTirta, the idea for T-Shirt sizes is just a tool for estimation. Translators would get rewarded based on the estimated size of the task. The XS mainly was Just an example for useful gets started tasks. I think @zicozibu example makes some good points on compensation as an agreeable structure from what was initially proposed.

The Badges do add an extra level of complexity, but it was an idea that we can implement later on in development. I don't suggest we throw badges into the mix just yet but certainly start thinking about them we are comfortable estimating, translating and rewarding. 

I don't follow what you mean by the simple difficulty levels. Again the t-shirt sizes are just estimates used to judge the size of tasks. Adding difficulty levels would complicate things here. So best stick to the small, medium and large sizes 🤗

**Reply by zicozibu** [reply]
@PardusEidolon the difficulty levels is introduced to address this part of your explanation:

> We should keep in mind that word count is only indicative. A very technical proposal short in word count can take much more effort than a high word-count proposal that is 100% free of jargon. Likewise, budgets will dictate what we can spend on translation and language.

In essence, to make it fair for everyone to work on what seems to be the same t-shirt size, but actually differ on the technicalities (difficulty levels). So the reward would be just about the same for translating:
a.  a very technical short in word, and
b.  a high word-count proposal that is 100% free of jargon

**Reply by YanTirta** [reply]
I would really love to go over this in Monday's meeting session. We are still very much aligned @PardusEidolon and @zicozibu . Simply put what Andy is trying to say is; the use of small, medium and large size equates to difficulty levels. For an example :

A translation item is worth a 200 word count, which falls into the Category S; [S: 5 ADA (approximately 200 to 400 words)]. But since it has very technical terms, it is then bumped up into the Category M.

**So if we were to make it a pipeline, it would go like this :**
1. determine word count and initial size
2. bump the size up or down depending on the difficulty level to translate
